src/lib.rs

```diff
@@ -14,8 +15,77 @@ pub fn eval(
         println!("{}", e.pp(&lexer, &calc_y::token_epp));
     }
     match res {
-        Some(Ok(r)) => Ok(format!("Result: {:?}", r)),
-        // TODO: prvide more infomraiton about the exact error.
+        Some(x) => {
+            return Ok(format!("Result: {}", Eval::new(input).eval(&x)));
+        }
         _ => Err(String::from("Unable to evaluate expression.")),
     }
 }
+
+struct Eval<'a> {
+    s: &'a str,
+}
+
+// stolen from: https://github.dev/softdevteam/grmtools
+impl<'a> Eval<'a> {
+    fn new(s: &'a str) -> Self {
+        return Eval { s };
+    }
+
+    fn eval_r_exp(&self, nodes: &Vec<Node<DefaultLexeme, u32>>) -> i64 {
+        if nodes.len() == 1 {
+            return self.eval(&nodes[0]);
+        } else {
+            debug_assert_eq!(nodes.len(), 3);
+            return self.eval(&nodes[0]) + self.eval(&nodes[2]);
```

Comment:
Does this assume addition then?

Reply:
I think think this code assumes: `R_TERM` = addition and `R_FACTOR` = multiplication
Because of `calc.y`:
```
Expr: Expr '+' Term
    | Term ;

Term: Term '*' Factor
    | Factor ;

```
I might be wrong :)

Reply:
OK